ppp_hal/requesthandler.py

```diff
@@ -40,22 +40,88 @@ def query(query, fields):
         mc.set(key, r, time=Config().memcached_timeout)
     return r
 
+PAPER_FIELDS = ('abstract_s', 'releasedDate_s', 'modifiedDateY_i',
+        'uri_s', 'halId_s', 'title_s', 'authFullName_s', 'arxivId_s',
+        'authFirstName_s', 'authLastName_s', 'authOrganism_s',
+        'labStructName_s', 'version_i', 'language_s')
+
+def graph_from_paper(paper):
+    same_as = list(filter(bool, (
+        paper.get('arxivId_s', None),
+        )))
+    organizations = paper.get('authOrganism_s', []) + \
+            paper.get('labStructName_s', [])
+    authors = [{'@type': 'Person',
+                'name': fullname,
+                'givenName': firstname,
+                'familyName': lastname,
+               }
+               for (fullname, firstname, lastname) in
+               zip(paper['authFullName_s'], paper['authFirstName_s'],
+                   paper['authLastName_s'])]
+    d = {
+            '@type': 'ScholarlyArticle',
+            '@context': 'http://schema.org',
+            'description': paper.get('abstract_s', None),
+            'datePublished': paper['releasedDate_s'],
+            'dateModified': paper['modifiedDateY_i'],
+            '@id': paper['uri_s'],
+            'sameAs': same_as,
+            'name': paper['title_s'],
+            'sourceOrganization': [
+                {'@type': 'Organization',
+                 'name': x,
+                } for x in organizations],
+            'version': paper.get('version_i', None),
+            'author': authors,
+            'inLanguage': paper['language_s'],
+            }
+    d = {x: y for (x, y) in d.items() if y is not None}
+    return d
+
+def paper_resource_from_paper(paper):
+    paper_graph = graph_from_paper(paper)
+    paper_graph['author'] = [
+            {'@type': 'Person',
+             '@context': 'http://schema.org',
+             '@id': author, # TODO: Use an actual ID
```

Comment:
Aaaarg. (@Tpt is dead of a cardiac arrest)

Are you sure you want "John Do" as @id?